pkg/cli/cli.go

```diff
@@ -1232,6 +1244,126 @@ func runConnectionWebrtcAgentCommand(flags *ConnectFlags, c *command, args []str
 	return nil
 }
 
+func runConnectionWebSocketAgentCommand(flags *ConnectFlags, c *command, args []string, opts *subCommandOpts) error {
+	// Change Web URL to WebSocket URL
+	// e.g. https://127.0.0.1:1443/ -> wss://127.0.0.1:1443
+	wsRoot := flags.ServiceURL
+	wsRoot = strings.ReplaceAll(wsRoot, "https://", "wss://")
+	wsRoot = strings.ReplaceAll(wsRoot, "http://", "ws://")
+	if strings.HasSuffix(wsRoot, "/") {
+		wsRoot = wsRoot[:len(wsRoot)-1]
+	}
+
+	// Open local TCP port that ADB server connects to and make ADB server connect
+	// to the port
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		c.PrintErrf("Failed to listen ADB socket: %v", err)
+		return err
+	}
+	adbPort := l.Addr().(*net.TCPAddr).Port
+	go func() {
+		if err := opts.ADBServerProxy.Connect(adbPort); err != nil {
+			c.PrintErrf("Failed to connect ADB to device: %v\n", err)
+		}
+	}()
+	tcpConn, err := l.Accept()
+	if err != nil {
+		c.PrintErrf("Failed to accept ADB socket: %v", err)
+		return err
+	}
+	defer tcpConn.Close()
+
+	// Send connect result to the parent
+	result := ConnStatus{
+		ADB: ForwarderState{
+			Port:  adbPort,
+			State: StateAsStr(FwdConnected),
+		},
+	}
+	output, err := json.Marshal(result)
+	if err != nil {
+		return err
+	} else {
+		c.Println(string(output))
+	}
+
+	// Close all standard IOs to make agent as daemon
+	if cin, ok := c.InOrStdin().(io.Closer); ok {
+		cin.Close()
+	}
+	if cout, ok := c.OutOrStdout().(io.Closer); ok {
+		cout.Close()
+	}
+	if cerr, ok := c.ErrOrStderr().(io.Closer); ok {
+		cerr.Close()
+	}
```

Comment:
Would this output be used? In normal cases, the function wouldn't be terminated.

Reply:
This output is used by parent cvdr process of websocket_agent to show adb serial of the connection.

The parent process waits for websocket_agent's processes closing its standard IOs while making it as daemon.